service/resource/cloudformation/delete.go

```diff
@@ -3,14 +3,76 @@ package cloudformation
 import (
 	"context"
 
+	"github.com/aws/aws-sdk-go/aws"
+	awscloudformation "github.com/aws/aws-sdk-go/service/cloudformation"
+	"github.com/giantswarm/aws-operator/service/key"
+	"github.com/giantswarm/microerror"
 	"github.com/giantswarm/operatorkit/framework"
 )
 
 func (r *Resource) ApplyDeleteChange(ctx context.Context, obj, deleteChange interface{}) error {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+	deleteStackInput, err := toDeleteStackInput(deleteChange)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	stackName := deleteStackInput.StackName
+	if *stackName != "" {
+		_, err := r.awsClient.DeleteStack(&deleteStackInput)
```

Comment:
Why should the stack name decide when to process some action? I feel like the delete stack input type should be checked as a whole because this is the actual content being used to do the deletion. I want to prevent that some name and the real value diverges.

Reply:
In fact the only field in the `DeleteStackInput` struct that determines what to delete is the stack name https://github.com/aws/aws-sdk-go/blob/master/service/cloudformation/api.go#L5274 

If we pass an empty string here we will have a validation error, have a roundtrip to the server and need to check for it later, with the additional ugly error like this:
```
An error occurred (ValidationError) when calling the DeleteStack operation: 2 validation errors detected: Value '' at 'stackName' failed to satisfy constraint: Member must have length greater than or equal to 1; Value '' at 'stackName' failed to satisfy constraint: Member must satisfy regular expression pattern: [a-zA-Z][-a-zA-Z0-9]*|arn:[-a-zA-Z0-9:/._+]*
```
which would need a brittle matcher method in errors.go with a big FIXME in the comment.
IMO we can prevent all this with a simple check for an empty string, let me know wdyt.

Reply:
Ok cool. Then it is what it is I guess.